File: aeds02/tps/tp1/Arquivo.c
#include<math.h>
#include<stdio.h>
#include<stdlib.h>

int main() {
	FILE *p = fopen("exemplo.txt", "wb");
	int numEntradas, index;
	double tmp;
	
	scanf("%d", &numEntradas);
	
	index = 0;
	while(index < numEntradas) {
		scanf("%lf", &tmp);
		fwrite(&tmp, sizeof(double), 1, p);
		index++;
	}
	fclose(p);
	
	FILE *p2 = fopen("exemplo.txt", "rb");
	index = numEntradas;
	while(index > 0) {
		fseek(p2, (index * 8) - 8, SEEK_SET);
		fread(&tmp, sizeof(double), 1, p);
		
		if(tmp - (int)tmp == 0.0) {
			printf("%d\n", (int)tmp);
		} else {
			printf("%0.3lf\n", tmp);
		}
		
		index--;
	}

	return 0;
}

File: aeds02/tps/tp1/Arquivo.java
import java.io.*;
import java.lang.Math;

class Arquivo {
	public static void imprimir(double entrada) {
		if(entrada - Math.floor(entrada) == 0.0) {
			MyIO.println((int)entrada);
		} else {
			MyIO.println(entrada);
		}
	}

	public static void main(String[] args) throws Exception {
		RandomAccessFile raf = new RandomAccessFile("exemplo.txt", "rw");
		int numEntradas = MyIO.readInt();
		int index = 0;
		while(index < numEntradas) {
			double tmp = MyIO.readDouble();
			raf.writeDouble(tmp);
			index++;
		}
		raf.close();
		
		RandomAccessFile raf2 = new RandomAccessFile("exemplo.txt", "r");
		index = numEntradas;
		while(index > 0) { 
			raf2.seek((8 * index) - 8);
			imprimir(raf2.readDouble());
			index--;
		}
		raf2.close();
		
		
	}
}
